src/compiler/moduleNameResolver.ts

```diff
@@ -2896,7 +2902,7 @@ function loadModuleFromSpecificNodeModulesDirectory(extensions: Extensions, modu
 
     const loader: ResolutionKindSpecificLoader = (extensions, candidate, onlyRecordFailures, state) => {
         let pathAndExtension =
-            loadModuleFromFile(extensions, candidate, onlyRecordFailures, state) ||
+            (rest || !(state.features & NodeResolutionFeatures.EsmMode)) && loadModuleFromFile(extensions, candidate, onlyRecordFailures, state) ||

```

Comment:
Can you point me to trace that chnged or you added for this. i couldnt find it.. probably missed it.

Reply:
These are the tests in nodeNextModuleResolution1 and 2. 1, especially, shows that in EsmMode, there has to be a `rest` -- that is, a subdirectory inside node_modules -- in order to call loadModeFromFile. That means a file like /a/node_modules/foo.d.ts doesn't show up in the trace.

CommonJS traces are not supposed to change, so I reverted the hundreds of trace changes from earlier.